lib/main.coffee

```diff
@@ -37,47 +28,79 @@ module.exports =
 
   configSub: null
   editorSub: null
-  watching: Object.create null
+  buffers: new Map
   debug: false
 
   activate: (state) ->
     return if atom.inSpecMode() or not atom.inDevMode()
 
     @configSub = atom.config.observe 'grammar-live-reload.enabled', (enabled) =>
-      return @editorSub?.dispose() unless enabled
 
-      reload = @reload.bind this
-      @editorSub = atom.workspace.observeTextEditors (editor) =>
+      unless enabled
+        @editorSub?.dispose()
+        @editorSub = null
+        @buffers.forEach (subs) -> subs.dispose()
+        @buffers.clear()
+        return
+
+      resolvePackageName = (grammarPath) ->
+        packPath = path.resolve grammarPath, '../..'
+        for pack in atom.packages.getActivePackages()
+          return pack.name if pack.path is packPath
+
+      isGrammarPath = (filePath) ->
```

Comment:
extract the function

Reply:
It's not used outside that scope, though. It's better to keep it close to where it is used, because minor perf improvements to config changes are not a necessity.